src/dogecoin.cpp

```diff
@@ -75,6 +75,51 @@ unsigned int CalculateDogecoinNextWorkRequired(const CBlockIndex* pindexLast, in
     return bnNew.GetCompact();
 }
 
+bool CheckAuxPowProofOfWork(const CBlockHeader& block, const Consensus::Params& params)
+{
+    /* Except for legacy blocks with full version 1, ensure that
+       the chain ID is correct.  Legacy blocks are not allowed since
+       the merge-mining start, which is checked in AcceptBlockHeader
+       where the height is known.  */
+    if (!block.nVersion.IsLegacy() && params.fStrictChainId && block.nVersion.GetChainId() != params.nAuxpowChainId)
```

Comment:
At the moment chainID does not get set on blocks if I read it correctly. At least on regtest, newly generated blocks will have nVersion = 3 so this verification fails on regtest. We should either set the CURRENT_VERSION in CBlock to one with the correct chainID or call SetChainId on the newly created block. getauxblock handles this afaik so this should be merely a regtest issue (testnet has fStrictChainId = false).